sklearn/decomposition/online_lda.py

```diff
@@ -0,0 +1,668 @@
+"""
+
+=============================================================
+Online Latent Dirichlet Allocation with variational inference
+=============================================================
+
+This implementation is modified from Matthew D. Hoffman's onlineldavb code
+Link: http://www.cs.princeton.edu/~mdhoffma/code/onlineldavb.tar
+"""
+
+# Author: Chyi-Kwei Yau
+# Author: Matthew D. Hoffman (original onlineldavb implementation)
+
+import numpy as np
+import scipy.sparse as sp
+from scipy.special import gammaln
+
+from ..base import BaseEstimator, TransformerMixin
+from ..utils import (check_random_state, check_array,
+                     gen_batches, gen_even_slices)
+from ..utils.validation import NotFittedError
+
+from ..externals.joblib import Parallel, delayed, cpu_count
+from ..externals.six.moves import xrange
+
+from ._online_lda import (mean_change, _dirichlet_expectation_1d,
+                          _dirichlet_expectation_2d)
+
+
+def check_non_negative(X, whom):
```

Comment:
If this is a copy from NMF, it should probably be moved to ``sklearn.utils.validation`` and imported in both places.